tests/source/issue-5488.rs

```diff
@@ -0,0 +1,11 @@
+// rustfmt-use_field_init_shorthand: true
+fn main() {
+    // Since MyStruct is a tuple struct, it should not be shorthanded to
+    // MyStruct { 0 } even if use_field_init_shorthand is enabled.
+    let instance = MyStruct { 0: 0 };
```

Comment:
Whether MyStruct is a tuple struct or not isn't represented in the AST, and isn't obvious just by looking at the code snippet. Could you add the tuple struct definition for MyStruct before main to make it 100% clear that MyStruct is a tuple struct.